source/guide_ejabberd.rst

```diff
@@ -132,81 +129,170 @@ Provide location of your keys and certificates for TLS transport:
       ip: "::"
       module: ejabberd_c2s
-      max_stanza_size: 262144
-      shaper: c2s_shaper
-      access: c2s
-      starttls_required: true
+      …
     - # 'secure' c2s
       port: <port-2>
       ip: "::"
       module: ejabberd_c2s
       tls: true
-      max_stanza_size: 262144
-      shaper: c2s_shaper
-      access: c2s
+      …
     - # s2s
       port: <port-3>
       ip: "::"
       module: ejabberd_s2s_in
-      max_stanza_size: 524288
+      …
     - # 'secure' s2s
       port: <port-4>
       ip: "::"
       module: ejabberd_s2s_in
       tls: true
-      max_stanza_size: 524288
+      …
+
+DNS Records
+---------------
+Since standard ports cannot be used on Uberspace DNS records must be set for c2s
+and s2s connections. Refer to the `XMPP wiki <https://wiki.xmpp.org/web/SRV_Records>`_
+for setup and point them to the corresponding ports.
+
+::
+
+ _xmpp-client._tcp.isabell.example. 86400 IN SRV 5 0 <port-1> isabell.example.
+ _xmpps-client._tcp.isabell.example. 86400 IN SRV 4 0 <port-2> isabell.example.
+ _xmpp-server._tcp.isabell.example. 86400 IN SRV 5 0 <port-3> isabell.example.
+ _xmpps-server._tcp.isabell.example. 86400 IN SRV 4 0 <port-4> isabell.example.
+
+File Transfer Proxy
+-------------------
+
+Configure ``mod_proxy65`` by setting the ``host`` and ``port`` values:
 
-Disable TLS for HTTP as it is going to be provided through web backends and disable the last two listeners:
+.. code-block:: ini
+  :emphasize-lines: 6,9
+
+  modules:
+    …
+    mod_proxy65:
+      access: local
+      max_connections: 5
+      host: "proxy.isabell.example"
+      name: "File Transfer Proxy"
+      ip: "::"
+      port: <port-5>
+    …
+
+Web-based Features
+------------------
+
+For web-based features like file uploads and web sockets adjust the
+configuration to listen on port 5443 without TLS:
 
 .. code-block:: ini
- :emphasize-lines: 5,13-23
+  :emphasize-lines: 7
 
+  listen:
+    …
     -
       port: 5443
       ip: "::"
       module: ejabberd_http
-    #  tls: true
+      #tls: true
       request_handlers:
         "/admin": ejabberd_web_admin
         "/api": mod_http_api
         "/bosh": mod_bosh
         "/captcha": ejabberd_captcha
         "/upload": mod_http_upload
         "/ws": ejabberd_http_ws
-    #-
-    #  port: 5280
-    #  ip: "::"
-    #  module: ejabberd_http
-    #  request_handlers:
-    #    "/admin": ejabberd_web_admin
-    #-
-    #  port: 1883
-    #  ip: "::"
-    #  module: mod_mqtt
-    #  backlog: 1000
+    …
+
+Additionally create a web-backend for ``xmpp.isabell.example/`` on port 5443.
 
-Add your admin user:
+.. include:: includes/web-backend.rst
+
+HTTP File Upload
+----------------
+
+Configure the ``put_url`` and the ``doc_root`` settings for ``mod_http_upload``
+to match the domain and your user folder:
 
 .. code-block:: ini
- :emphasize-lines: 2-4
+  :emphasize-lines: 4,10
+
+  modules:
+    …
+    mod_http_upload:
+      put_url: "https://xmpp.@HOST@/upload"
```

Comment:
Should this be `xmpp.` or `upload.`? https://github.com/Uberspace/lab/commit/b0158b24cb07f2f06df09f795a9d22c929ce5d9f

Reply:
I like to use `xmpp.` because—in this guide—it is used for all HTTP-based features like BOSH, web sockets, file uploads etc. and not only uploads.